lib/src/upZenohClient.cpp

```diff
@@ -42,6 +46,14 @@ std::shared_ptr<UpZenohClient> UpZenohClient::instance(void) noexcept {
         }
 
         handle = std::make_shared<UpZenohClient>(ConstructToken());
+        auto rpc_handle = static_pointer_cast<ZenohRpcClient>(handle);
+        //  providing rpc client info after instance constructed is an error
+        if (src_authority && src_entity && w_handle.use_count() > 0) return nullptr;
+        // if constructing, not providing both authority and entity is an error
+        if (!src_authority) return nullptr;
+        if (!src_entity) return nullptr;
+        rpc_handle->clientAuthority = *src_authority;
+        rpc_handle->clientEntity = *src_entity;
```

Comment:
It's better to check for the existence of the source parameters before constructing an instance using make_shared. Additionally, I would recommend passing the source parameters through to the `ZenohRpcClient` as constructor parameters instead of relying on pointer casting and setting parameters inside of another class. Lastly, the check on line 51 will never fail as we do not enter this block when `w_handle.use_count() > 0`.

I recommend changing this code to this:

```suggestion
        // When constructing a new instance for the first time, these parameters must be provided.
        if (!src_authority || !src_entity) {
            return nullptr;
        }
        
        handle = std::make_shared<UpZenohClient>(ConstructToken(), *src_authority, *src_entity);
```

If you want to check that `src_authority` and `src_entity` are not set when an instance already exists, that check will need to be done right before line 39 and line 45.